components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -686,15 +690,15 @@ List<Object> listUsers(Map<String, Boolean> requiredAttributes, Integer offset,
                 log.debug(message);
             }
         } else {
-            List<Object> scimUsers = getUserDetails(coreUsers, requiredAttributes);
+            scimUsers = getUserDetails(coreUsers, requiredAttributes);
             if (totalUsers != 0) {
-                users.set(0, Math.toIntExact(totalUsers)); // Set total number of results to 0th index.
```

Comment:
change totalusers to int and check for null and 0

Reply:
Many other methods use long so discussed and decided to keep as long instead of changing to int.